src/dbnode/encoding/m3tsz/iterator.go

```diff
@@ -37,39 +37,30 @@ import (
 type readerIterator struct {
 	is   encoding.IStream
 	opts encoding.Options
-	tess encoding.TimeEncodingSchemes
-	mes  encoding.MarkerEncodingScheme
 
 	// internal bookkeeping
-	t   time.Time     // current time
-	dt  time.Duration // current time delta
-	vb  uint64        // current float value
-	xor uint64        // current float xor
-	err error         // current error
+	tsIterator TimestampIterator
+	vb         uint64 // current float value
```

Comment:
Why no XORIterator to match the XOREncoder?

Reply:
I re-implemented some of that logic in the proto iterator because I didn't want to re-use the really confusing logic from the m3tsz one. They share the `ReadXOR` helper function though

Reply:
Do the two implementations essentially do the same thing? Just seems a little odd cause we have pairs of encoders and iterators.